include/gsl/string_span

```diff
@@ -94,7 +94,7 @@ namespace details
     template <class CharT>
     std::ptrdiff_t string_length(const CharT* str, std::ptrdiff_t n)
     {
-        if (str == nullptr || n <= 0) return 0;
+        Expects(str != nullptr && n > 0);
```

Comment:
This looks close to me, but not quite following the spirit of the original issue that was raised. I think it's ok for `str` to be `nullptr` and in that case, `string_length` should return `0`. Similarly, the `n` can legitimately be `0`, regardless of whether or not `str` is `nullptr`. This change would disallow those cases.

I think what you want to prevent here is when `n < 0` (which should never happen) and when `str == nullptr && n > 0` which would also be problematic.